test/provider/AppDirectory.unittest.ts

```diff
@@ -0,0 +1,109 @@
+import 'jest';
+import 'reflect-metadata';
+
+import {AppDirectory} from '../../src/provider/model/AppDirectory';
+import {Application} from '../../src/client/directory';
+
+
+enum StorageKeys {
+    URL = 'url',
+    APPLICATIONS = 'applications'
+}
+
+const appTemplate: Application = {
+    appId: 'id',
+    name: 'test-app',
+    manifest: '',
+    manifestType: '',
+    intents: [
+        {
+            name: 'Call',
+            contexts: [
+                'dial'
+            ],
+            customConfig: {}
+        }
+    ]
+};
+
+// Generate applications
+const generateApplications = (amount: number): Application[] => {
+    return Array(amount).fill(appTemplate).map((app, i) => ({
+        ...app,
+        appId: `id-${i}`,
+        name: `${app.name}-${i}`
+    }));
+};
+
+describe('AppDirectory Unit Tests', () => {
+    let appDirectory: AppDirectory;
+    let store: LocalForage;
+
+    afterEach(async () => {
+        await store.clear();
+    });
+
+    describe('Storage', () => {
+        beforeEach(async () => {
+            appDirectory = new AppDirectory();
+        });
+
+        afterEach(async () => {
+            await store.clear();
+        });
+
+        it('Correctly sets up the store', async () => {
+            store = appDirectory['_store'];
+            // await store.ready();
+            const url = await store.getItem(StorageKeys.URL);
+            const applications = await store.getItem(StorageKeys.APPLICATIONS);
+            expect(url).toEqual(null);
+            expect(applications).toEqual(null);
+        });
+
+        describe('Storing fetchData', () => {
+            it('Correctly sets applications in the store', async () => {
+                appDirectory['fetchData'] = jest.fn(() => Promise.resolve([]));
+                await appDirectory.getAllApps();
+                const applications = await store.getItem(StorageKeys.APPLICATIONS);
```

Comment:
Does this work?.. `store` at this point would be an object that was created in the previous test run. Seems like it would always be empty (since it's cleared in the afterEach, and then never re-populated)?

The implementation also relies on the `'Correctly sets up the store'` test to run first - you wouldn't be able to use `.only` or `.skip` with these tests.

Every test should be fully independent - would make sense to me to add a `beforeEach` at the same level as the `store.clear()` `afterEach` that creates the directory and extracts the store object. Not sure exactly where `localForage` is saving the data in this case - but if it's `localStorage`, the test could also query the store directly rather than extracting the private field from `appDirectory`.